### ISSUE TITLE
Join GitHub Pre-Release Program

### ISSUE THREAD

**Issue by mikeal** [nodejs/TSC#271]
GitHub has a system for early access to experimental GitHub features.

My personal account is already on it but I also have the option of enrolling orgs. Would we like to see features before they are complete so that we can give GitHub feedback? We did this with the new orgs a few years ago and it went quite well.

**Comment by nebrius** [comment]
For some reason I thought we were still enrolled in this. I'm up for trying it out!

**Comment by mikeal** [comment]
The org beta program was a one-off, this is a longer term beta for all kinds of features.

**Comment by jasnell** [comment]
SGTM

**Comment by ChALkeR** [comment]
@mikeal I don't have a vote here, but I will share my two cents.

Judging by the thread, I am not convinced that it gives the Node.js org any gain.
What specific features are we missing that would be provided by this integration?
I do not understand the positive impact yet, and it is not outlined in this proposal.

The negative impact would be:
 * Potentially less tested (and potentially les secure) features. I find the probability of vulnerabilites in pre-release features more significant than in production. Moreover, that is one of the points of the pre-release program, afaik.
 * I doubt that we would be able to follow the ToS of the pre-release program.

Check the [Terms of Service](https://help.github.com/articles/github-pre-release-program/#2-confidentiality-access-to-pre-release-materials), they state:

> As a pre-release program participant, you may get access to special information that isn’t available to the rest of the world. Due to the sensitive nature of this information, it’s important for us to make sure that you keep that information secret.
> 
> You agree that any non-public information we give you, or that you may have access to as a pre-release program participant, will be considered GitHub’s confidential information (collectively, “Confidential Information”), regardless of whether it is marked or identified as such. 

We have a _large number_ of members in the org, are they all going to be informed of these ToS? I find it pretty probable that someone from the org members would break this point of the ToS without even knowing it. E.g. post a screenshot on Twitter, or something like that.

So far, I'm mostly -1 on this (not that my vote matters, though :wink: ).

_And yes, I personally opted in pretty long ago._

**Comment by mikeal** [comment]
@ChALkeR I don't quite share the ToS concerns. There won't be non-public information shared with people who don't sign the ToS. Also, I would be the only person *signing* the ToS when I add this support to the org.

As for the concerns about untested features, that is a risk, but there's also a risk that GitHub will add features without our feedback that negatively impact our project. We steered them away from certain problems in the orgs beta. We're also seeing some unintended consequences from the reviews feature, which we did not test or give feedback into.

We're one of the largest and most active projects on GitHub so we have the most to lose from features that negatively impact our workflows. Our experience with the orgs beta was that features don't land in the beta that are broken, so I don't think there's a big risk that we will lose productivity in that way.
